### PR TITLE
feat: add new util function (getRequest)

### PR DESCRIPTION
**Pull request by hideokamoto**
The `getRequest()` function helps us to make a skill by using TypeScript.
We can get a specific type of the request object. 

## Description
Add a new util function `getRequest`.
To get request object with own expected type.

## Motivation and Context
Related: Missing typescript definitions? #398

### Currently
We can get the request object by the code.

```typescript
const { request} = handlerInput.requestEnvelope
```

But the object has several type, so we can not get any property without using `any`.

```typescript
// ERROR
const state = handlerInput.requestEnvelope.request.dialogState

-> Property 'dialogState' does not exist on type 'PlaybackFinishedRequest'.

// PASS
const state = (handlerInput.requestEnvelope.request as IntentRequest).dialogState
```

But these coding style is not comfortable and hard to understand for a new developer.

### Solution

The `getRequest` function is my solution.
It can update the `request` object type by using TypeScript Generics.

```typescript
// For Intent request
const request = getRequest<IntentRequest>(handlerInput.requestEnvelope)
const state = request.dialogState

// For AudioPlayer intent request
const req = getRequest<interfaces.audioplayer.PlaybackFinishedRequest>(intentRequestEnvelope)
const sec = req.offsetInMilliseconds
```

## Testing

```
$ lerna --scope ask-sdk-core run gulp

  113 passing (101ms)


=============================== Coverage summary ===============================
Statements   : 100% ( 357/357 )
Branches     : 98.99% ( 98/99 )
Functions    : 96.33% ( 105/109 )
Lines        : 100% ( 336/336 )
================================================================================

[00:17:25] Finished 'test' after 2.59 s
[00:17:26] Finished 'tsc' after 3.58 s
[00:17:26] Finished 'default' after 3.6 s
lerna success run Ran npm script 'gulp' in 1 package in 5.4s:
lerna success - ask-sdk-core
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs(Add new document content)
- [ ] Translate Docs(Translate document content)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Review by ShenChen93 — COMMENTED** [review]
Thanks @hideokamoto for the PR.  It seems you haven't apply our tslint rules for the test case. Could you apply our rule and send another revision ? Thanks

**Comment by hideokamoto** [comment]
@ShenChen-Amazon 
Thanks.
I've fixed it today :)

File: ask-sdk-core/tst/util/RequestEnvelopeUtil.spec.ts [hunk 1]
```diff
@@ -87,6 +88,20 @@ describe('RequestEnvelopeUtils', () => {
         }).to.throw(`Expecting request type of IntentRequest but got LaunchRequest.`);
     });
 
+    it('should return the intent request object', () => {
+        const request = getRequest<IntentRequest>(intentRequestEnvelope)
```
Comment (ShenChen93) [inline ask-sdk-core/tst/util/RequestEnvelopeUtil.spec.ts:92]: The CR looks good. However, It seems you still haven't applied our tslint rule here. I rewrite this test case:
```ts
it('should return the intent request object', () => {
        const request = getRequest<IntentRequest>(intentRequestEnvelope);
        expect(request).deep.eq({
            type: 'IntentRequest',
            requestId: null,
            timestamp: null,
            locale: null,
            intent: {
                confirmationStatus: null,
                name: 'MockIntent',
                slots: {
                    mockSlot: {
                        confirmationStatus: null,
                        name: 'mockSlot',
                        value: 'mockSlotValue',
                        resolutions: null,
                    },
                },
            },
            dialogState: 'STARTED',
        });
    });
```
Could you please use it in your new revision ? Thanks!

Comment (hideokamoto) [reply ask-sdk-core/tst/util/RequestEnvelopeUtil.spec.ts:92]: Thanks.
I've fixed it :)
https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/pull/582/commits/ae0e324d8c3aa8b5c7ff63c7fa016f82724405e9

Comment (hideokamoto) [reply ask-sdk-core/tst/util/RequestEnvelopeUtil.spec.ts:92]: BTW.

Probably, the `gulp tslint` does not referenced the tst dir.
So I made a new issue about the problem. -> tslint dose not check tst dir files #585

Thanks.

**Review by ShenChen93 — APPROVED** [review]

**Review by RonWang — APPROVED** [review]

**Pull request merged**